src/rust/engine/src/nodes.rs

```diff
@@ -1579,8 +1584,17 @@ impl NodeError for Failure {
       path[path_len - 1] += " <-"
     }
     throw(&format!(
-      "Dependency graph contained a cycle:\n  {}",
-      path.join("\n  ")
+      "The dependency graph contained a cycle:\
+      \n\n  \
+      {}\
+      \n\n\
+      If the dependencies in the above path are for your BUILD targets, you may need to use more \
+      granular targets or replace BUILD target dependencies with file dependencies. If they are \
+      not for your BUILD targets, then please file a Github issue!\
```

Comment:
Could we inspect the `path` to check if `transitive_target` is in it? If so, show the original error message we had. Otherwise, show the "please file a GitHub issue" piece.

The Python test should already handle regressions if, say, we were to rename that `@rule` so the string match is no longer accurate.